### PR TITLE
Return response of initial HEAD request

### PR DESCRIPTION
**Pull request by konstin**
We need this response to determine our caching strategy, specifically we want to know if the response is immutable.

I'm not completely happy with that caching strategy yet but it works for pypi and wheels at a url should be immutable (otherwise you get a hash mismatch when deploying).

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -110,7 +110,7 @@ impl AsyncHttpRangeReader {
             CheckSupportMethod::NegativeRangeRequest(initial_chunk_size) => {
                 Self::new_tail_request(client, url, initial_chunk_size).await
```
Comment (baszalmstra) [inline src/lib.rs:111]: Would it make sense to also return the response for this request?

File: src/lib.rs [hunk 2]
```diff
@@ -204,10 +204,10 @@ impl AsyncHttpRangeReader {
         })
     }
 
-    async fn new_head(
+    pub async fn new_head(
         client: reqwest::Client,
         url: reqwest::Url,
-    ) -> Result<Self, AsyncHttpRangeReaderError> {
+    ) -> Result<(Self, reqwest::Response), AsyncHttpRangeReaderError> {
```
Comment (baszalmstra) [inline src/lib.rs:210]: Maybe it makes sense to only return the headers of the request?

**Comment by konstin** [comment]
> This looks good to me but wouldn't you then also want to add additional headers to the initial request?

yes, i'll likely add that in the future, though i still have to figure out how to integrate this; i'm still looking for something that implements http caching semantics on top of reqwest without being a middleware, https://github.com/06chaynes/http-cache/issues/57). For pypi i get away with just checking if the cache control header says "immutable".

**Comment by baszalmstra** [comment]
Yeah I would like to have something like that too! Im following your issue. :)

Would you be able to look at the other comments I left?

**Review by baszalmstra — APPROVED** [review]

**Pull request merged**